libbeat/outputs/elasticsearch/client.go

```diff
@@ -183,9 +186,9 @@ func (client *Client) Publish(batch publisher.Batch) error {
 // PublishEvents sends all events to elasticsearch. On error a slice with all
 // events not published or confirmed to be processed by elasticsearch will be
 // returned. The input slice backing memory will be reused by return the value.
-func (client *Client) publishEvents(
-	data []publisher.Event,
-) ([]publisher.Event, error) {
+func (client *Client) publishEvents(ctx context.Context, data []publisher.Event) ([]publisher.Event, error) {
+	span, ctx := apm.StartSpan(ctx, "publishEvents", "output")
```

Comment:
I think it would be better if we didn't create this span, and just relied on the "publish" transaction, but this is probably necessary for now if we really want the labels.

Ideally, the code for recording the labels (`events_original`, etc.) should be done in a single location, rather than in specific output implementations. This would at lease give high-level information for all outputs. Specific outputs could optionally extend those spans with output-specific information, or add sub-spans (e.g. the apmelasticsearch span).

I think to do that we'd need to change the Batch and/or Observer interfaces (to report batch-specific stats) and/or the Publish signature (to return the stats).